cpu/stm32_common/include/periph_cpu_common.h

```diff
@@ -244,6 +244,28 @@ typedef struct {
     uint8_t bus;                    /**< APB bus */
 } pwm_conf_t;
 
+/**
+ * @brief   QDEC channel
+ */
+typedef struct {
+    gpio_t pin;             /**< GPIO pin mapped to this channel */
+    uint8_t cc_chan;        /**< capture compare channel used */
+} qdec_chan_t;
+
+/**
+ * @brief   QDEC configuration
+ */
+typedef struct {
+    TIM_TypeDef *dev;               /**< Timer used */
+    uint32_t max;
+    uint32_t rcc_mask;              /**< bit in clock enable register */
+    qdec_chan_t chan[TIMER_CHAN];    /**< channel mapping, set to {GPIO_UNDEF, 0}
```

Comment:
Aren't quadratic decoder supposed to have only 2 channels ?

Reply:
True ! Thank you.

It remains from pwm driver as I used it as a template. ;)
It is corrected on my side by defining a new macro QDEC_CHAN.

I will push it once tested.

Reply:
In the datasheet of the nucleo-f401 I noticed that it is possible to have a third channel for an absolute positioning, to zero the counter when this reference position is hit.

Reply:
Yes, it's possible, but does this channel would be managed the same way as the 2 others ?

Reply:
Using encoder Z index could lead to manage a different mode which could be not available on all architectures.

We could also consider other modes than X1,X2,X4 if we consider falling/rising edge, inverted signals, etc...
But I think it would turn it into something too much complex so I tried to keep it simple and "generic"